jetty-core/jetty-websocket/jetty-websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -206,18 +206,60 @@ public void onFrame(Frame frame, Callback coreCallback)
                 coreCallback.failed(new WebSocketException(endpointInstance.getClass().getSimpleName() + " FRAME method error: " + cause.getMessage(), cause));
                 return;
             }
+
+            switch (frame.getOpCode())
+            {
+                case OpCode.TEXT ->
+                {
+                    if (textHandle == null)
+                        autoDemand();
+                }
+                case OpCode.BINARY ->
+                {
+                    if (binaryHandle == null)
+                        autoDemand();
+                }
+                case OpCode.CONTINUATION ->
+                {
+                    if (activeMessageSink == null)
+                        autoDemand();
+                }
+                case OpCode.PING ->
+                {
+                    if (pingHandle == null)
+                        autoDemand();
+                }
+                case OpCode.PONG ->
+                {
+                    if (pongHandle == null)
+                        autoDemand();
+                }
+                case OpCode.CLOSE ->
+                {
+                    // Do nothing.
+                }
+                default ->
+                {
+                    coreCallback.failed(new IllegalStateException());
+                    return;
+                }
+            };
```

Comment:
I don't like putting this logic here.... I think similar behaviour can be implemented in the individual methods which can do something like:
```java
        if (xyzHandle != null)
        {
            xyzHandler.invoke(...);
            callback.succeeded();
            autoDemand();
        }
        else 
        {
            callback.succeeded();
            if (frameHandle == null)
                internalDemand();
            else
                autoDemand();
        }
```